src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -41,7 +42,7 @@ public class ScriptHost : JobHost
         private Action<FileSystemEventArgs> _restart;
         private AutoRecoveringFileSystemWatcher _scriptFileWatcher;
         private AutoRecoveringFileSystemWatcher _debugModeFileWatcher;
-        private int _directoryCountSnapshot;
+        private ImmutableList<string> _directorySnapshot; 
```

Comment:
Curious, any reason you chose to go with an `ImmutableList<string>` vs an `ImmutableArray<string>` as mentioned in my previous comment? Unless there's a specific reason I'm missing, based on how this is being used, `ImmutableArray<string>` would be a better fit here.